core/src/main/java/com/yahoo/oak/Slice.java

```diff
@@ -76,21 +52,28 @@ void duplicateBuffer() {
      * Allocation info and metadata setters
      * ------------------------------------------------------------------------------------*/
     // Reset all not final fields to invalid state
-    void invalidate() {
-        blockID     = NativeMemoryAllocator.INVALID_BLOCK_ID;
-        reference   = invalidReferenceValue;
-        length      = UNDEFINED_LENGTH_OR_OFFSET;
-        offset      = UNDEFINED_LENGTH_OR_OFFSET;
-        buffer      = null;
-        associated  = false;
-    }
+    abstract void invalidate();
```

Comment:
Could we add a default implementation that invalidates the relevant fields to avoid code duplication?

Reply:
I have added default in the abstract Slice and in the SliceSyncRecycle I overwrite the invalidate() method